Sem. 01/Pract. 01/Task08.cpp

```diff
@@ -1,93 +1,191 @@
-#include <iostream>
+//task8 from oop practicum 1
 
-const int MAX_SIZE_NAME = 30;
-const int MAX_GROUP_SIZE = 30;
-
-struct Student {
-	char name[MAX_SIZE_NAME + 1];
-	unsigned int fn;
-	double grade;
-};
-
-struct StudentGroup {
-	int amountOfStudents;
-	double averageGrade;
-	Student groupArr[MAX_GROUP_SIZE];
-};
-
-double findSum(const Student* arr, int numberOfStudents) {
-	double sum = 0;
-	for (int i = 0; i < numberOfStudents; i++) {
-		sum += arr[i].grade;
+#include <iostream>
+using std::cout;
+using std::cin;
+using std::swap;
+
+const int MAX_NAME_SIZE = 999;
+
+namespace Students 
+{
+	struct Student {
+		char name[MAX_NAME_SIZE];
+		double averageScore;
+	};
+
+	void readStudent(Student& student) {
+		cin >> student.name;
+		cin >> student.averageScore;
 	}
-
-	return sum;
 }
 
-void initGroupOfStudents(StudentGroup& group) {
-	for (int i = 0; i < group.amountOfStudents; i++) {
-		std::cin >> group.groupArr[i].fn;
-		std::cin >> group.groupArr[i].name;
-		std::cin.ignore();
-		std::cin >> group.groupArr[i].grade;
-	}
+namespace Groups {
+	using namespace Students;
 
-	group.averageGrade = findSum(group.groupArr, group.amountOfStudents) / group.amountOfStudents;
-}
+	struct Group
+	{
+		int32_t countStudents;
+		Student* students;
+		double averageScore;
+	};
+
+	double calculateAverageGroupScore(Student* students, const int32_t& countStudents) {

```

Comment:
const Student*, do not put & on primitive types - & takes between 4 and 8 bytes, which is more than any primitive (except the double in the case &=4bytes)